File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        
        if head==None or head.next==None:
            return False
        
        slow=head
        fast=head
        
        while(True):
            if(fast.next!=None):
                fast=fast.next
            else:
                return False
            
            if(fast.next!=None):
                fast=fast.next
            else:
                return False
            
            slow=slow.next
            
            if fast==slow:
                return True
            
        
        return True
        
        
        

File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(!head || head->next==nullptr)
          return false;
      
      int count=0;
      ListNode* t=head;
      
      while(count<10000){
        t=t->next;
        count++;
        if(t==nullptr)
          return false;
      }
      
      return true;
    }
};